services/src/main/java/org/keycloak/protocol/oid4vc/issuance/keybinding/JwtCNonceHandler.java

```diff
@@ -80,10 +79,10 @@ public String buildCNonce(List<String> audiences, Map<String, Object> additional
         RealmModel realm = keycloakSession.getContext().getRealm();
         final String issuer = OID4VCIssuerWellKnownProvider.getIssuer(keycloakSession.getContext());
         // TODO discussion about the attribute name to use
-        final Integer nonceLifetimeMillis = realm.getAttribute(OID4VCIConstants.C_NONCE_LIFETIME_IN_SECONDS, 60);
+        final Integer nonceLifetimeSeconds = realm.getAttribute(OID4VCIConstants.C_NONCE_LIFETIME_IN_SECONDS, 60);
         audiences = Optional.ofNullable(audiences).orElseGet(Collections::emptyList);
-        final Instant now = Instant.now();
-        final long expiresAt = now.plus(nonceLifetimeMillis, ChronoUnit.SECONDS).getEpochSecond();
+        final long nowSeconds = Time.currentTime();
+        final long expiresAt = nowSeconds + nonceLifetimeMillis;
```

Comment:
I was expecting you would rename with the IDE...
```suggestion
        final long expiresAt = nowSeconds + nonceLifetimeSeconds;
```